src/Extensions/IAsyncEnumerableExtensions.cs

```diff
@@ -2265,7 +2265,95 @@ private static async Task _enumerate(AsyncEnumerator<TSource>.Yield yield, Disti
 
             return false;
         }
-        
+
+#endregion
+
+#region Count
+        /// <summary>
+        /// Returns the number of elements in a sequence.
+        /// </summary>
+        /// <param name="source">An <see cref="IAsyncEnumerable{T}"/> that contains the elements to apply the predicate to.</param>
+        /// <param name="cancellationToken">A cancellation token to cancel the async operation.</param>
+        /// <typeparam name="TSource">The type of the elements of <paramref name="source"/>.</typeparam>
+        /// <returns>The number of elements in the input sequence.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> or <paramref name="predicate"/> is null.</exception>
+        public static async Task<int> CountAsync<TSource>(
+            this IAsyncEnumerable<TSource> source,
+            CancellationToken cancellationToken = default)
+        {
+            if (source == null)
+                throw new ArgumentNullException(nameof(source));
+
+            var enumerator = source.GetAsyncEnumerator(cancellationToken);
+
+            int count = 0;
+
+            try
+            {
+                while (true)
+                {
+                    if (!await enumerator.MoveNextAsync().ConfigureAwait(false))
+                    {
+                        break;
+                    }
+
+                    count++;
+                }
+            }
+            finally
+            {
+                await enumerator.DisposeAsync().ConfigureAwait(false);
+            }
+
+            return count;
+        }
+
+        /// <summary>
+        /// Returns a number that represents how many elements in the specified sequence satisfy a condition.
```

Comment:
[nit] I'd match the language between the overloads - either a number that represents or the number of (the later seems more natural to me but either seems correct)

Reply:
Was copying MSDocs. But definitely better lol.
![image](https://user-images.githubusercontent.com/5658563/79227762-649fa180-7e2e-11ea-9795-f2c46aa36513.png)